nilearn/mass_univariate/tests/test_permuted_least_squares.py

```diff
@@ -140,49 +195,38 @@ def test_permuted_ols_check_h0_noeffect_labelswap(random_state=0):
     assert_array_less(np.diff(all_mse_intercept2.mean(1)), 0)
 
 
-def test_permuted_ols_check_h0_noeffect_signswap(random_state=0):
+def test_permuted_ols_check_h0_noeffect_signswap(random_state=RANDOM_STATE):
+    """Check that h0 is close to the theoretical distribution \
+    for permuted OLS with sign swap.
+
+    Theoretical distribution is known for this simple design \
+        (= t(n_samples - dof)).
+    """
     rng = check_random_state(random_state)
 
     # design parameters
-    n_samples = 100
+    n_samples = N_SAMPLES
     n_regressors = 1
 
     # create dummy design with no effect
     target_var = rng.randn(n_samples, 1)
     tested_var = np.ones((n_samples, n_regressors))
 
-    # permuted OLS
-    # We check that h0 is close to the theoretical distribution, which is
-    # known for this simple design (= t(n_samples - dof)).
-    perm_ranges = [10, 100, 1000]  # test various number of permutations
-    all_kstest_pvals = []
-
     # we compute the Mean Squared Error between cumulative Density Function
     # as a proof of consistency of the permutation algorithm
     all_mse = []
+
+    # test various number of permutations
+    perm_ranges = [10, 100, 1000]
+    all_kstest_pvals = []
     for i, n_perm in enumerate(np.repeat(perm_ranges, 10)):
-        pval, orig_scores, h0 = permuted_ols(
-            tested_var,
-            target_var,
-            model_intercept=False,
-            n_perm=n_perm,
-            two_sided_test=False,
-            random_state=i,
-        )
-        assert h0.shape == (n_regressors, n_perm)
+        h0 = permuted_ols_no_intercept(tested_var, target_var, n_perm, i)
+        ############################
+        # CHECK if the original value of DoF was correct
+        df = n_samples
+        ############################
         h0_intercept = h0[0, :]
-
-        # Kolmogorov-Smirnov test
-        kstest_pval = stats.kstest(h0_intercept, stats.t(n_samples).cdf)[1]
-        all_kstest_pvals.append(kstest_pval)
-        mse = np.mean(
-            (
-                stats.t(n_samples).cdf(np.sort(h0_intercept))
-                - np.linspace(0, 1, h0_intercept.size + 1)[1:]
-            )
-            ** 2
-        )
```

Comment:
Maybe I am missing something but shouldn't the degrees of freedom in here be `n_samples - 1` ?

I changed it in my refactoring but maybe there is something I am missing...

Reply:
I think you're right.